### PR TITLE
feat(channel): Add new overwrite helper methods

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

Guys don't code when you're sick

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [ ] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [ ] For the documentation
  - [x] To add a new feature
  - [ ] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

File: interactions/api/models/channel.py [hunk 1]
```diff
@@ -1870,6 +1882,67 @@ async def get_permissions_for(self, member: "Member") -> Permissions:
...
+    async def add_permission_overwrites(self, overwrites: Iterable[Overwrite]) -> "Channel":
+        """
+        .. versionadded:: 4.4.0
+
+        Add multiple overwrites to the channel.
+
+        :param Iterable[Overwrite] overwrites: The overwrites to add to the channel.
+        """
+
+        ow = self.permission_overwrites or []
```
Comment (Damego) [inline interactions/api/models/channel.py]: Rename ´ow´ to ´_overwrites´

**Review by Damego — COMMENTED** [review]
Missed return and rtype in the docstrings

File: interactions/api/models/channel.py [hunk 2]
```diff
@@ -1870,6 +1882,73 @@ async def get_permissions_for(self, member: "Member") -> Permissions:
 
         return Permissions(permissions)
 
+    async def add_permission_overwrite(
+        self,
+        id: Union[int, str, Snowflake, User, Role],
+        type: Optional[Literal[0, 1, "0", "1"]] = None,
+        allow: Optional[Union[int, Permissions, str]] = None,
+        deny: Optional[Union[int, Permissions, str]] = None,
```
Comment (Damego) [inline interactions/api/models/channel.py]: Why here is None, not MISSING?

File: interactions/api/models/channel.py [hunk 3]
```diff
@@ -1870,6 +1882,109 @@ async def get_permissions_for(self, member: "Member") -> Permissions:
...
+        :param Union[int, str, Snowflake, User, Role] id: The ID of the User/Role to create the overwrite on.
+        :param Optional[Literal[0, 1, "0", "1"]] type: The type of the overwrite. 0 for Role 1 for User.
+        :param Optional[Union[int, Permissions, str]] allow: Permissions to allow
+        :param Optional[Union[int, Permissions, str]] deny: Permissions to deny
+        :param Optional[str] reason: The reason to be shown in the audit log
+        :return: The updated channel
+        :rtype: Channel
+        """
+
+        if not deny and not allow:
```
Comment (Damego) [inline interactions/api/models/channel.py:1907]:
```suggestion
        if deny is MISSING and allow is MISSING:
```

**Review by Damego — COMMENTED** [review]
I think this better

File: interactions/api/models/channel.py (hunk 3 above)
Comment (sillyElvy) [reply interactions/api/models/channel.py:1907]: that makes it worse

MISSING is not (`__bool__`). So it we check for missing only the user could just pass `None` for both

Comment (Damego) [reply interactions/api/models/channel.py:1907]: Okay then

**Comment by Damego** [comment]
Does it work with new changes?

**Comment by sillyElvy** [comment]
> Does it work with new changes?

on my machine yep

**Review by Damego — APPROVED** [review]

**Pull request merged**